diff_drive_controller/doc/userdoc.rst

```diff
@@ -24,3 +24,135 @@ Other features
+
+~/cmd_vel_out []
+
+
+
+
+Publishers
+,,,,,,,,,,,
+~/odom []
+
+/tf
+
+
+Services
+,,,,,,,,,
+
+
+Parameters
+------------
+
+Wheels
+,,,,,,,
+left_wheel_names [array<string>]
+  Link names of the left side wheels.
+
+right_wheel_names  [array<string>]
+  Link names of the right side wheels.
+
+wheel_separation [double]
+  Shortest distance between the left and right wheels.
+  If this parameter is wrong, the robot will not behave correctly in curves.
+
+wheels_per_side [integer]
+  Number of wheels on each wide of the robot.
+  This is important to take the wheels slip into account when multiple wheels on each side are present.
+  If there are more wheels then control signals for each side, you should enter number or control signals.
+  For example, Husky has two wheels on each side, but they use one control signal, in this case "1" is the correct value of the parameter.
+
+wheel_radius [double]
+  Radius of a wheel, i.e., wheels size, used for transformation of linear velocity into wheel rotations.
+  If this parameter is wrong the robot will move faster or slower then expected.
+
+wheel_separation_multiplier [double]
+  Correction factor for wheel separation (TODO(destogl): Please help me describe this correctly)
+
+left_wheel_radius_multiplier [double]
+  Correction factor when radius of left wheels differs from the nominal value in ``wheel_radius`` parameter.
+
+right_wheel_radius_multiplier [double]
+  Correction factor when radius of right wheels differs from the nominal value in ``wheel_radius`` parameter.
+
+Odometry
+,,,,,,,,,
+base_frame_id [string] (default: "base_link")
+  Name of the frame for odometry.
+  This frame is parent of ``base_frame_id`` when controller publishes odometry.
+
+odom_frame_id [string] (default: "odom")
+  Name of the robot's base frame that is child of the odometry frame.
+
+pose_covariance_diagonal [array<double>[6]] (default: "[0.0, 0.0, 0.0, 0.0, 0.0, 0.0]")
+  Odometry covariance for the encoder output of the robot for the pose.
+  These values should be tuned to your robot's sample odometry data, but these values are a good place to start:
+  ``[0.001, 0.001, 0.001, 0.001, 0.001, 0.01]``.
+
+twist_covariance_diagonal [array<double>[6]] (default: "[0.0, 0.0, 0.0, 0.0, 0.0, 0.0]")
+  Odometry covariance for the encoder output of the robot for the speed.
+  These values should be tuned to your robot's sample odometry data, but these values are a good place to start:
+  ``[0.001, 0.001, 0.001, 0.001, 0.001, 0.01]``.
+
+open_loop [bool] (default: "false")
+  If set to false the odometry of the robot will be calculated from the commanded values and not from feedback.
+
+position_feedback [bool] (default: "true")
+  Is there position feedback from hardware.
+
+enable_odom_tf [bool] (default: "true")
+  Publish transformation between ``odom_frame_id`` and ``base_frame_id``.
+
+velocity_rolling_window_size [int] (default: "10")
+  (TODO(destogl): Please help me describe this correctly)
+
+Commands
```

Comment:
this is slightly misleading: on first impression I thought these are the topics to command the controller, the rst heading should help it though when rendered